3Dmol/glmodel.js

```diff
@@ -1446,14 +1446,24 @@ $3Dmol.GLModel = (function() {
          * @param {number} framenum - model's atoms are set to this index in frames list
          */
         this.setFrame = function(framenum) {
+           
             if (frames.length == 0) {
                 return;
             }
-            if (framenum >= 0 && framenum < frames.length) {
-                atoms = frames[framenum];
-            }
-            else {
-                atoms = frames[frames.length-1];
+            if (framenum < 0 || framenum > frames.length) {
+                framenum = frames.length - 1;
+            }
+            atoms = frames[framenum];
```

Comment:
Each frame should be dynamically setup when the frame is requested.  Or even better, the coordinates should be read into the existing atoms.  This means the check for the url should come first, although you still need to check the number of frames.